tests/functional/object/mcg/test_noobaa_db_pg_expansion.py

```diff
@@ -0,0 +1,88 @@
+import logging
+
+from ocs_ci.utility import utils
+from ocs_ci.framework.pytest_customization.marks import (
+    vsphere_platform_required,
+    skipif_mcg_only,
+    tier2,
+    red_squad,
+    mcg,
+)
+from ocs_ci.ocs.resources.pod import get_ceph_tools_pod
+from ocs_ci.ocs.resources.pvc import get_pvc_objs, get_pvc_size
+from ocs_ci.ocs.exceptions import CommandFailed, CephToolBoxNotFoundException
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.ocp import OCP
+
+logger = logging.getLogger(__name__)
+
+
+@mcg
+@red_squad
+@skipif_mcg_only
+@vsphere_platform_required
+class TestNoobaaDbPgExpansion:
+    """
+    Test noobaa DB PG expansion
+    Cleanup operation is not applicable due to shrink volume size on PVC is not supported in kubernetes
+    """
+
+    @tier2
+    def test_noobaa_db_pg_expansion(self, scale_noobaa_db_pod_pv_size):
+        """
+        Test to check noobaa db pg PVC is getting expanded using ceph block pool.
+        """
+
+        # Check ceph health is reporting HEALTH_OK or not
+        utils.ceph_health_check(tries=20)
+
+        # record MAX avail size of ceph block pool before expanding capacity
+
+        try:
+            ceph_toolbox = get_ceph_tools_pod(
+                namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+            )
+        except (AssertionError, CephToolBoxNotFoundException) as ex:
+            raise CommandFailed(ex)
+
+        size_in_byte = ceph_toolbox.exec_ceph_cmd(
+            ceph_cmd="ceph df",
+            timeout=120,
+        )
+        for pool in size_in_byte["pools"]:
+            if pool.get("name") == constants.DEFAULT_BLOCKPOOL:
+                max_avai_bytes = pool["stats"]["max_avail"]
+                avail_size = int(max_avai_bytes) / constants.GB
+                break
+        logger.info(f"Max avail size on ceph block pool is {avail_size} GB")
+
+        # record PVC capacity before expanding capacity
+        noobaa_db_pvc_obj = get_pvc_objs(pvc_names=[constants.NOOBAA_DB_PVC_NAME])[0]
+        pvc_capacity = get_pvc_size(noobaa_db_pvc_obj)
+        logger.info(f"Current PVC capacity is {pvc_capacity} GB")
+
+        # Validate PVC expansion against available RBD size and current PVC size
```

Comment:
@udaysk23 why this step is neccessary?

Reply:
@mashetty330 Script is increasing Capacity by averaging current size and available RBD size. Also script is using current pvc capacity to validate the real change happened or not

Reply:
sorry i referenced the wrong code

Reply:
```python
        assert (
            pvc_capacity <= avail_size
        ), f"PVC expand operation is not valid, current PVC capacity {pvc_capacity} >= ceph RBD AVAIL size {avail_size}"

        # Calculate new PVC size by averaging MAX AVAIL of the RBD Pool and current PVC size
        new_pvc_size = round((avail_size + pvc_capacity) / 2)
```

Reply:
This piece of code i meant to ask. Is it neccessary to use average of available size and current pvc capacity? 
because for example, if avail size 400GB and pvc size is 50GB and the average of that would be 225GB for only noobaa db pv and we know that we cant revert this back also. Cant we just increase it by small amount? like maybe add half of what current pv size is. something like new_pvc_size = current_size + (current_size//2)

Reply:
Thanks for the suggestion @mashetty330. I have added one more condition to get valid PVC value